src/main/java/com/mparticle/kits/OneTrustKit.java

```diff
@@ -12,16 +32,90 @@
+
+        final Map<String, String> consentMapping = new HashMap<String, String>();
+
+        if (mobileMappingValues != null) {
+            try {
+                // Convert mobileMappingValues and extract new consentMapping
+                // value -> OneTrust Cookie Value
+                // map -> mParticle Consent Purpose
+                JSONArray consentJSONArray = new JSONArray(mobileMappingValues);
+                for (int i = 0; i < consentJSONArray.length(); i++) {
+                    JSONObject consentJSONObject = consentJSONArray.optJSONObject(i);
+
+                    consentMapping.put(consentJSONObject.getString("value"), consentJSONObject.getString("map"));
+                }
+
+            } catch (JSONException jsonException) {
+                Logger.warning(jsonException, "MP OT JSON ERROR");
+            }
+        }
+
+        // Listen for One Trust Consent changes and create ne mPartcle Consent
+        // Changes based on these events
+        // intent.getAction -> OT cookie name
+        // intent.getIntExtras(OTBroadcastServiceKeys.EVENT_STATUS) -> (bool-int) consented?
+        categoryReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                Log.i("BroadcastService", "MP OT Intent name: " + intent.getAction() +
+                        " status = " + intent.getIntExtra(OTBroadcastServiceKeys.EVENT_STATUS, -1));
+
+                OneTrustKit.this.createConsentEvent(intent.getAction(), intent.getIntExtra(OTBroadcastServiceKeys.EVENT_STATUS, -1));
+            }
+        };
+
+        for(final String consentElement: consentMapping.keySet()) {
+            // Register receiver from above based on cookie value dispatched by OneTrust SDK
+            context.registerReceiver(categoryReceiver, new IntentFilter(consentElement));
+
+            // Fetch Consent Status from OneTrust based on cookie value
+            new OTPublishersHeadlessSDK(context).getConsentStatusForGroupId(consentElement);
+
+            // Dispatch creation of initial consent state till after init is done
+            new Handler().post(new Runnable() {
+                @Override
+                public void run() {
+                    OneTrustKit.this.createConsentEvent(consentMapping.get(consentElement), 0);
+                }
+            });
+        }
+
+
         // Save mapping to disk. Will be retrieved by OneTrust Mobile SDK
         saveToDisk(mobileMappingValues);
         return null;
     }
 
+    // Create an mParticle consent state based on One Trust mapping
+    // Status is an integer that probably should be a boolean
+    //  1 = Consent Given
+    //  0 = Consent Not Given
+    // -1 = Consent has not been collected/ sdk is not yet initialized
+    private void createConsentEvent(String purpose, Integer status) {
+        MParticleUser user = MParticle.getInstance().Identity().getCurrentUser();
```

Comment:
This needs to be null-checked and deferred (if null). Check the PR I just put up. https://github.com/mparticle-integrations/mparticle-android-integration-onetrust/pull/6